core/src/main/java/org/apache/iceberg/util/DecimalUtil.java

```diff
@@ -31,7 +31,7 @@ private DecimalUtil() {}
   public static byte[] toReusedFixLengthBytes(
       int precision, int scale, BigDecimal decimal, byte[] reuseBuf) {
     Preconditions.checkArgument(
-        decimal.scale() == scale,
+        decimal.scale() <= scale,
```

Comment:
Is this safe? It is re-use fix length bytes but in this case we aren't using all the bytes.